ElementX/Sources/Screens/JoinRoomScreen/View/JoinRoomScreen.swift

```diff
@@ -148,27 +148,80 @@ struct JoinRoomScreen: View {
     }
     
     @ViewBuilder
-    var buttons: some View {
+    var bottomContent: some View {
         switch context.viewState.mode {
-        case .loading, .unknown:
+        case .loading:
             EmptyView()
-        case .knock:
+        case .joinable:
+            joinButton
+        case .unknown, .restricted: // If unknown, do our best.
+            VStack(spacing: 24) {
+                bottomNoticeMessage(L10n.screenJoinRoomJoinRestrictedMessage)
+                
+                joinButton
+            }
+        case .knockable:
             Button(L10n.screenJoinRoomKnockAction) { context.send(viewAction: .knock) }
                 .buttonStyle(.compound(.super))
         case .knocked:
             Button(L10n.screenJoinRoomCancelKnockAction) { context.send(viewAction: .cancelKnock) }
                 .buttonStyle(.compound(.secondary))
-        case .join:
-            Button(L10n.screenJoinRoomJoinAction) { context.send(viewAction: .join) }
-                .buttonStyle(.compound(.super))
+        case .inviteRequired:
+            bottomNoticeMessage(L10n.screenJoinRoomInviteRequiredMessage)
         case .invited:
             ViewThatFits {
                 HStack(spacing: 8) { inviteButtons }
                 VStack(spacing: 16) { inviteButtons }
             }
+        case .banned(let sender, let reason):
+            if let sender, let reason {
+                bottomErrorMessage(title: L10n.screenJoinRoomBanByMessage(sender),
+                                   subtitle: L10n.screenJoinRoomBanReason(reason))
+            } else {
+                bottomErrorMessage(title: L10n.screenJoinRoomBanMessage, subtitle: nil)
+            }
         }
     }
     
+    func bottomNoticeMessage(_ notice: String) -> some View {
+        HStack(alignment: .top, spacing: 12) {
+            CompoundIcon(\.info)
+            
+            Text(notice)
+                .font(.compound.bodyLGSemibold)
+                .foregroundStyle(.compound.textPrimary)
+            
+            Spacer()
+        }
```

Comment:
Could we
a) use a `Label` here (otherwise you need to add a `.accessibilityHidden` on the icon.
b) use a frame rather than a spacer

E.g. something like this:
```suggestion
        Label(notice, icon: \.info)
            .labelStyle(.custom(alignment: .top, spacing: 12)
            .frame(maxWidth: .infinity, alignment: .leading)
```

Reply:
Sure is, thanks for the tip!